File: 219.ContainsDuplicateII/duplicate.cc
class Solution {
public:
    bool containsNearbyDuplicate(vector<int>& nums, int k) {
        map<int,int> m;
        for (int i = 0; i < int(nums.size()); i++) {
            map<int,int>::iterator it = m.find(nums[i]);
            if (it != m.end()) {
                int index = m[nums[i]];
                if (i - index <= k) {
                    return true;
                }
            }
            m[nums[i]] = i;
        }

        return false;
    }
};

File: 219.ContainsDuplicateII/duplicate.py
class Solution(object):
    def containsNearbyDuplicate(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: bool
        """
        d = {}
        for i in range(len(nums)):
            if nums[i] in d:
                index = d[nums[i]]
                if i - index <= k:
                    return True
            d[nums[i]] = i
        return False
